### PR TITLE
Avoiding NPE - informative exception if master connection == null

### PR DESCRIPTION
**Pull request by denis-aes**
I've accidentally connected to mongo slave instead of master and got an NPE. After this spent some time debugging to understand NPE reasons.

This exception will make code more informative.

### CODE REVIEW

File: lib/src/database/db.dart [hunk 1]
```diff
@@ -209,7 +209,12 @@ class Db {
       }
 
       if (connection == null) {
-        connection = _masterConnection;
+        if (_masterConnection != null) {
+          connection = _masterConnection;
+        }
+        else {
+          throw new MongoDartError('No master connection');
```
Comment (Pacane) [inline lib/src/database/db.dart]: Can you extract this message into a constant. And also, you repeat this exact piece of code 3 times. You could extract a method from that and call that instead.

Comment (Pacane) [inline lib/src/database/db.dart]: I don't know if it'd be easy to write a test for this as well, but if you think it's not too hard, I'd like to see one in there. Thanks for the contribution btw.

**Comment by denis-aes** [comment]
@Pacane I've updated change. This looks much better. What do you think?

File: lib/src/database/db.dart [hunk 2]
```diff
@@ -209,7 +210,7 @@ class Db {
       }
 
       if (connection == null) {
-        connection = _masterConnection;
+        connection = _masterConnectionVerified;
```
Comment (Pacane) [inline lib/src/database/db.dart]: I think I'd wrap the null check as well. And maybe call it `isMasterConnectionValid` or something like that.

Comment (Pacane) [inline lib/src/database/db.dart]: (The null check for `connection`)

Comment (Pacane) [inline lib/src/database/db.dart]: Yeahhh.. thinking about it, you're doing as assignment, I don't like queries that have side effects. Leave it like this.

Comment (denis-aes) [inline lib/src/database/db.dart]: @Pacane, sorry, I didn't understand you. Final resolution - leave everything like it is now?

Comment (Pacane) [inline lib/src/database/db.dart]: Yes like it is right now.

Comment (denis-aes) [inline lib/src/database/db.dart]: @Pacane I can make test, if you help to mock slave connection.
What do I need at the moment for successful merge to master?

Comment (Pacane) [inline lib/src/database/db.dart]: Just bump the version of the package as a minor and I'll merge. I believe it's gonna be hard to write a test for this after all.

Comment (denis-aes) [inline lib/src/database/db.dart]: @Pacane done.

Comment (denis-aes) [inline lib/src/database/db.dart]: @Pacane thank you!)

**Pull request merged**